apps/site/assets/ts/schedule/components/line-diagram/SingleStop.tsx

```diff
@@ -229,7 +232,12 @@ const SingleStop = ({
           <div className="m-schedule-diagram__stop-name">
             <a href={`/stops/${routeStop.id}`}>
               <h4>
-                {MaybeAlert(stopAlerts)} {routeStop.name}{" "}
+                {MaybeAlert(stopAlerts)}
+                {query ? (
```

Comment:
Based on how `MatchHighlight` is implemented, I don't think this conditional is needed? If there is no query, it just displays the text with no highlight.

Reply:
Slightly different issue -- the `query` prop expects a string, but here it can potentially be undefined (as `query` is an optional prop on this particular component). I could change the input to `query || ""` but it seems silly to include the unneeded `<MatchHighlight />` -- and allowing `undefined` to be passed in seems meh to me. Any suggestions?

Reply:
My suggestion would be:

```jsx
<h4>
  {MaybeAlert(stopAlerts)}
  <MatchHighlight text={routeStop.name} query={query} />
</h4>
```

And make `query` an optional property. This keeps all the highlighting-related logic, including the decision about whether to do any highlighting, encapsulated in `MatchHighlight`.

There are other ways, for sure — what I think is non-ideal here is that the "whether to do highlighting" decision is currently _split_ between `MatchHighlight` and its callers (it checks for empty strings, but callers have to check for `undefined`).